src/v/cluster/id_allocator_stm.h

```diff
@@ -74,7 +77,37 @@ class id_allocator_stm final : public raft::state_machine {
         int64_t state;
     };
 
-    ss::future<> apply(model::record_batch) final;
+    bool _should_cache{false};
+    bool _procesing_legacy{true};
+    model::offset _prepare_offset{0};
+    std::vector<allocation_cmd> _cache;
+
+    struct state_cmd {
+        static constexpr uint8_t record_key = 3;
+        int64_t next_state;
+    };
+
+    ss::future<stm_allocation_result>
+      do_allocate_id(model::timeout_clock::duration);
+    ss::future<bool> set_state(int64_t, model::timeout_clock::duration);
+
+    ss::future<> apply(model::record_batch) override;
+
+    ss::future<> apply_snapshot(stm_snapshot_header, iobuf&&) override;
+    ss::future<stm_snapshot> take_snapshot() override;
+    ss::future<> handle_eviction() override;
+    ss::future<bool> sync(model::timeout_clock::duration);
+
+    mutex _lock;
+
+    int64_t _batch_size;
+    int16_t _log_capacity;
+
+    int64_t _processed{0};
```

Comment:
would be great to add single line comment on top of variables definitions to indicate what are they. It can be read from code but is is a little bit of effort

Reply:
Good idea, I'll add a description of variables and the flow in general.

Reply:
+1, to be honest I don't understand what pid means in this context

Reply:
renamed pid to id and added a comment